src/store/actions.ts

```diff
@@ -299,5 +300,37 @@ export default {
 
   setRestorationExpiry ({ commit }, expiry: string) {
     commit('mutateRestorationExpiry', expiry)
+  },
+
+  setRestorationApprovalType ({ commit }, type: ApprovalTypes) {
+    commit('mutateRestorationApprovalType', type)
+  },
+
+  setStateFilingRestoration (context): Promise<any> {
+    return new Promise((resolve, reject) => {
+      LegalServices.fetchFiling(context.getters.getStateFilingUrl)
+        .then((response) => {
+          const stateFilingRestoration = response.restoration
+          // commit data to store
+          context.commit('mutateStateFilingRestoration', stateFilingRestoration)
+          // return the state filing restoration object
+          resolve(stateFilingRestoration)
+        })
+        .catch(error => {
+          reject(error)
+        })
+    })
+  },
+
+  setRestorationCourtOrder ({ commit }, courtOrder: CourtOrderIF): void {
+    commit('mutateRestorationCourtOrder', courtOrder)
+  },
+
+  setExpiryValid ({ commit }, valid: boolean): void {
+    commit('mutateExpiryValid', valid)
+  },
+
+  setApprovalTypeValid ({ commit }, valid: boolean): void {
+    commit('mutateApprovalTypeValid', valid)
```

Comment:
It doesn't look like [vue-property-decorator](https://www.npmjs.com/package/vue-property-decorator) has a `@Mutation` decorator. What's a clean way to declare an imported mutation?

Reply:
I hear you @severinbeauvais . I think the best solution is to start using the Composition API which not only has better support for typing but also helps in our transition to Vue3.

Reply:
Yes, we could start using the Composition API for new things. It is supported in Vue 2.7.10, which we are using.

If we can solve the mutation-calling issue (2 comments in this PR) in a day or less, let's do it. If not, ignore it for this component and we'll tackle it in another component, or globally in the future.

Reply:
Shall I try to fix that now in this PR? I don't think I'll be able to complete that by the end of the day however since this is uncharted territory for me.

Reply:
It's looking like "ignore it for now". @jonathan-longe OK with this?

Reply:
@JazzarKarim  I think we can use `@Mutation` like this:

```javascript
import { Action, Getter, Mutation } from 'vuex-class'

export class SomeComponent extends Vue {
@Mutation('foo') mutationFoo

// blah, blah, blah
}
```
See:  https://github.com/ktsn/vuex-class

Reply:
Alright then, on it!

Reply:
Looks great! I'm OK with this conversation (but it's not mine to resolve).